### ISSUE TITLE
Rom factory crashing on associations with version 4.0 of rom

### ISSUE THREAD

**Issue by cmavromoustakos** [rom-rb/rom-factory#11]
The association now crashes when run against the latest version of rom:

It appears that for associations the key that is being passed in to the 'def fetch(key)' method is the actual factory that has been initialized, instead of the key. 

The error:

```
NoMethodError:
  undefined method `to_sym' for #<IdentityProviders:0x007fe1c0554e58>
  Did you mean?  to_s
# /Users/cmavromoustakos/.rvm/gems/ruby-2.4.1/gems/rom-core-4.0.0/lib/rom/registry.rb:63:in `fetch'
# /Users/cmavromoustakos/.rvm/gems/ruby-2.4.1/gems/rom-factory-0.4.0/lib/rom/factory/dsl.rb:59:in `association'
# ./spec/support/factories.rb:12:in `block in <main>'
# /Users/cmavromoustakos/.rvm/gems/ruby-2.4.1/gems/rom-factory-0.4.0/lib/rom/factory/dsl.rb:31:in `initialize'
```

Debugger

```
From: /Users/cmavromoustakos/.rvm/gems/ruby-2.4.1/gems/rom-core-4.0.0/lib/rom/registry.rb @ line 66 ROM::Registry#fetch:

    60: def fetch(key)
    61:   raise ArgumentError.new('key cannot be nil') if key.nil?
    62:
    63:   require 'pry'
    64:   binding.pry
    65:
 => 66:   elements.fetch(key.to_sym) do
    67:     return yield if block_given?
    68:
    69:     raise self.class.element_not_found_error.new(key, self)
    70:   end
    71: end

[1] pry(#<ROM::RelationRegistry>)> key
=> #<IdentityProviders name=ROM::Relation::Name(identity_providers) dataset=#<Sequel::Postgres::Dataset: "SELECT \"identity_providers\".\"id\", \"identity_providers\".\"name\", \"identity_providers\".\"saml_metadata_url\", \"identity_providers\".\"created_at\", \"identity_providers\".\"updated_at\" FROM \"identity_providers\" ORDER BY \"identity_providers\".\"id\"">>


```

```
Factory.define(:user) do |f|
  ...
  f.association(:identity_provider)
  f.timestamps
end
```

Gemfile.lock
```
  ┆ rom (4.0.0)
  ┆ ┆ rom-changeset (~> 1.0)
  ┆ ┆ rom-core (~> 4.0)
  ┆ ┆ rom-mapper (~> 1.0)
  ┆ ┆ rom-repository (~> 2.0)
  ┆ rom-changeset (1.0.0)
  ┆ ┆ dry-core (~> 0.3, >= 0.3.1)
  ┆ ┆ rom-core (~> 4.0.0.rc)
  ┆ ┆ transproc (~> 1.0)
  ┆ rom-core (4.0.0)
  ┆ ┆ concurrent-ruby (~> 1.0)
  ┆ ┆ dry-container (~> 0.6)
  ┆ ┆ dry-core (~> 0.3)
  ┆ ┆ dry-equalizer (~> 0.2)
  ┆ ┆ dry-initializer (~> 2.0)
  ┆ ┆ dry-types (~> 0.12, >= 0.12.1)
  ┆ ┆ rom-mapper (~> 1.0)
  ┆ rom-factory (0.4.0)
  ┆ ┆ dry-configurable (~> 0.1)
  ┆ ┆ dry-core (~> 0.2)
  ┆ ┆ dry-struct (~> 0.2)
  ┆ ┆ faker (~> 1.7)
  ┆ rom-mapper (1.0.0)
  ┆ ┆ dry-core (~> 0.3, >= 0.3.1)
  ┆ ┆ dry-equalizer (~> 0.2)
  ┆ ┆ transproc (~> 1.0)
  ┆ rom-repository (2.0.0)
  ┆ ┆ dry-struct (~> 0.3)
  ┆ ┆ rom-mapper (~> 1.0)
  ┆ rom-sql (2.0.0)
  ┆ ┆ dry-core (~> 0.3)
  ┆ ┆ dry-equalizer (~> 0.2)
  ┆ ┆ dry-types (~> 0.12, >= 0.12.1)
      rom-core (~> 4.0)
  ┆ ┆ sequel (>= 4.49)
```

**Comment by solnic** [comment]
Please grab it from master for now. We'll release a new version soon.

**Comment by cmavromoustakos** [comment]
@solnic thanks, that was it.

**Comment by GustavoCaso** [comment]
I will close the issue them.

**Comment by solnic** [comment]
Just released rom-factory 0.5.0
